crates/criticalup-core/src/download_server_client.rs

```diff
@@ -93,32 +93,78 @@ impl DownloadServerClient {
         self.cache_dir.join("keys.json")
     }
 
-    fn product_release_manifest_cache_path(&self, product: &str, release: &str) -> PathBuf {
+    async fn product_release_manifest_cache_path(&self, product: &str, release: &str) -> PathBuf {
         self.product_release_cache_path(product, release)
+            .await
             .join("manifest.json")
     }
 
-    fn product_release_cache_path(&self, product: &str, release: &str) -> PathBuf {
-        self.cache_dir
+    async fn product_release_cache_path(&self, product: &str, release: &str) -> PathBuf {
+        let old_cache_dir = self.cache_dir.join("artifacts").join(product).join(release);
+
+        let new_cache_dir = self
+            .cache_dir
             .join("artifacts")
             .join("products")
             .join(product)
             .join("releases")
-            .join(release)
+            .join(release);
+
+        if old_cache_dir.exists() {
```

Comment:
A getter function like this is not an appropriate place for a migration to go. You should find a better place.